src/tink/syntaxhub/FrontendContext.hx

```diff
@@ -98,26 +98,36 @@ class FrontendContext {
     }
     return ret;
   }
-  
+
   static function moduleForType(name:String) {
-    if (name.indexOf('__impl') != -1 || plugins.getData().length == 0) return None;
+    if (name.indexOf('__impl') != -1 || plugins.getData().length == 0) return;
     var pack = name.split('.');
-    var name = pack.pop();
-    var actual = pack.concat(['__impl', name]).join('.');
-    var exists = 
+    var tname = pack.pop();
+    var actual = pack.concat(['__impl', tname]).join('.');
+    cache[name] = {
+      pack: pack,
+      name: tname,
+      pos: Context.currentPos(),
+      fields: [],
+      kind: TDAlias(actual.asComplexType())
+    }
+    var exists =
       try {
-        Context.getModule(actual);
+        Context.getType(actual);
         true;
       }
       catch (e:Dynamic) false;
-    
+
     if (!exists) {
-      var module = buildModule(pack, name);
-      if (module.types.length == 0) return None;
-      
+      var module = buildModule(pack, tname);
```

Comment:
At the moment, `buildModule` might see a `TDAlias` to a type that never ends up existing...

This is easy to fix (by simply inlining `buildModule` here and make the test run just after `seekFile` but before and `FrontendPlugin.parse` call), but I'm not sure it's necessary.